plugins/woocommerce/src/Internal/ProductFilters/QueryClauses.php

```
@@ -239,13 +304,96 @@ public function add_attribute_clauses( $args, $chosen_attributes ) {
 		return $args;
 	}
 
+	/**
+	 * Add query clauses for taxonomy filter (e.g., product_cat, product_tag).
+	 *
+	 * @internal For exclusive usage of WooCommerce core, backwards compatibility not guaranteed.
+	 *
+	 * @param array $args           Query args.
+	 * @param array $chosen_taxonomies {
+	 *     Chosen taxonomies array.
+	 *
+	 *     @type array {$taxonomy: Taxonomy name} {
+	 *         @type string[] $terms Chosen terms' slug.
+	 *     }
+	 * }
+	 * @return array
+	 */
+	public function add_taxonomy_clauses( array $args, array $chosen_taxonomies ): array {
+		if ( empty( $chosen_taxonomies ) ) {
+			return $args;
+		}
+
+		global $wpdb;
+
+		$tax_queries = array();
+
+		$all_terms = get_terms(
+			array(
+				'taxonomy' => array_keys( $chosen_taxonomies ),
+				'slug'     => array_merge( ...array_values( $chosen_taxonomies ) ),
+			)
+		);
+
+		if ( is_wp_error( $all_terms ) ) {
+			return $args;
+		}
```

Should we in some way log that there has been an error here? Even if we want everything to continue working in the frontend, `is_wp_error()` means something went wrong, no?

---

We can use our logger. Or we can throw an exception when WP_DEBUG is turned on, which one do you prefer?

---

Not a strong opinion, I lean more towards using the logger because if keeps things more similar independently of whether `WP_DEBUG` is true or not. But as mentioned, I don't have a strong opinion. :smile:

---

I'm okay on using WC_Logger here

---

Looking at the implementation of the `get_terms` function, it only returns WP_Error if some of the queried taxonomies are invalid. So instead of logging error, I think we can filter out the invalid taxonomy earlier. I don't see the value of logging this error now.